### ISSUE TITLE
Tab keyboard behaviour breaks when children composition changes

### ISSUE THREAD

**Issue by edleeks87** [Sage/carbon#4498]
### Current behaviour

When the children of `Tabs` change the refs are simply added to the existing `tabRefs` array rather than replacing them. This causes the keyboard navigation to break.

### Expected behaviour

Refs should be recalculated when the children composition changes so that keyboard navigation is not broken

### CodeSandbox or Storybook URL

https://codesandbox.io/s/carbon-tab-wrapper-latest-forked-0y542?file=/src/index.js

### Suggested Solution

Replace the implementation of `tabRefs` `tabs.component` (https://github.com/Sage/carbon/blob/208eb65d5cb7936bb9047d5ee5e6cfd70509c18d/src/components/tabs/tabs.component.js#L40) to `useMemo` implementation and refactor how the refs are applied to the TabTitle children.

```
const tabRefs = React.useMemo(
    () =>
      Array.from({ length: filteredChildren.length }).map(() =>
        React.createRef()
      ),
    [filteredChildren.length]
  );
``` 

Delete this function https://github.com/Sage/carbon/blob/208eb65d5cb7936bb9047d5ee5e6cfd70509c18d/src/components/tabs/tabs.component.js#L159 and replace this line https://github.com/Sage/carbon/blob/208eb65d5cb7936bb9047d5ee5e6cfd70509c18d/src/components/tabs/tabs.component.js#L221 with `ref={tabRefs[index]}`

### Carbon Version

94.2.0

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.
